test/src/com/xilinx/rapidwright/design/TestNetTools.java

```diff
@@ -0,0 +1,115 @@
+     *      get_nets -hier -parent_net -filter { TYPE == "GLOBAL_CLOCK" } 
+     * to retrieve all nets of type GLOBAL_CLOCK. 
+     * The method NetTools.isGlobalClockNet(Net net) returns true for these nets and false for others.
+     * 
+     * All Versal/Ultrascale+ designs under the RapidWrightDCP path will be checked, except for the following:
+     *      picoblaze4_ooc_X6Y60_X6Y65_X10Y60_X10Y65.dcp    The source cell bufgce_inst of this design is not placed yet.
+     *      picoblaze_ooc_X10Y235_unreadable_edif.dcp       This design would call vivado to generate an readable EDIF file.
+     *      two_clk_check_NetTools.dcp                      This design is not added to the RapidWrightDCP yet.
+     * 
+     * All Series-7 designs are not supported by NetTools.isGlobalClockNet(Net net) yet, so they are not tested now.
+     * @param pathAndlobalClockNames Use ' ' as the delimiter to split this string, where the first element is the path to the DCP file, 
+     *                               and the remaining elements are the names of the global clock nets reported by Vivado.
+     */
+    @ParameterizedTest
+    @CsvSource({
+            // UltraScale+
+            "bnn.dcp",
+            "hwct.dcp",
+            "bug701.dcp",
+            "hwct_pr1.dcp",
+            "inout.dcp",
+            "optical-flow.dcp",
+            "design_with_backslash_2021.2.dcp",
+            "design_with_backslash_2022.1.dcp",
+            "design_with_backslash_2022.2.dcp",
+            "gnl_2_4_3_1.3_gnl_3000_07_3_80_80_placed.dcp",
+            // "picoblaze4_ooc_X6Y60_X6Y65_X10Y60_X10Y65.dcp clk",          // The source cell bufgce_inst is not placed yet.
+            "picoblaze_ooc_X10Y235.dcp",
+            "picoblaze_ooc_X10Y235_2022_1.dcp",
+            // "picoblaze_ooc_X10Y235_unreadable_edif.dcp",                 // This case would call vivado to generate an readable EDIF file.
+            "picoblaze_partial.dcp clk",
+            "reduce_or_routed_7overlaps.dcp",
+            "testCopyImplementation.dcp",
+            
+            // UltraScale
+            "microblazeAndILA_3pblocks.dcp base_mb_i/clk_wiz_1/inst/clk_out1 base_mb_i/clk_wiz_1/inst/clkfbout_buf_base_mb_clk_wiz_1_0 base_mb_i/mdm_1/U0/No_Dbg_Reg_Access.BUFG_DRCK/Dbg_Clk_31 dbg_hub/inst/BSCANID.u_xsdbm_id/itck_i",
+            "microblazeAndILA_3pblocks_2024.1.dcp base_mb_i/clk_wiz_1/inst/clk_out1 base_mb_i/clk_wiz_1/inst/clkfbout_buf_base_mb_clk_wiz_1_0 base_mb_i/mdm_1/U0/No_Dbg_Reg_Access.BUFG_DRCK/Dbg_Clk_31 dbg_hub/inst/BSCANID.u_xsdbm_id/itck_i",
+
+            // Versal
+            "noc_tutorial_routed.dcp",
+            "picoblaze_2022.2.dcp",
+            "versal_cout_hq2.dcp",
+            // "two_clk_check_NetTools.dcp clk1_IBUF_BUFG clk2_IBUF_BUFG rst1 rst2",
+
+            // Series-7
+            // "bug226.dcp murax.apb3Router_1.io_mainClk",
+            // "bug349.dcp CLK_BUFG_BOT_R_X60Y48_BUFGCTRL_X0Y0_O",
```

Comment:
Since you have two designs with clocks here: is it much more work to support Series 7 too?

Reply:
I can make it support. Will do it in the incoming commit.

Reply:
> I can make it support. Will do it in the incoming commit.

Done.